### PR TITLE
add support for building a Link header

### PR DESCRIPTION
**Pull request by xadhoom**
I've split in 2 submodules, but kept the interface in ExLinkHeader, in order to have a "shorter" way to call the functions (and protecting the API from submodules changes).
Also tests have been split up.

I've put everything into a nested struct, using RFC5988 Registered Relation Types as keys.
This is a bit more verbose, but also more formal.

Parsing have been refactored to use the new struct, unfortunately breaking the current APIs.

There're also some integration test to check that header building is consistent with parsing.

Also Readme has been updated to reflect the change.

What do you think about that?

### CODE REVIEW

**Comment by simonrand** [comment]
Thanks for this @xadhoom – my time is a bit tight at the moment – I'll get through this all properly as soon as I can!

**Comment by simonrand** [comment]
Hey @xadhoom – took me a while to get to this – I reviewed on a recent flight, I'm going to merge this and make some changes before creating a new release – thanks for the work on this!

One tip in future – I've included Credo in the project – if you can run Credo (http://credo-ci.org / `$ mix credo`) and handle any issues that'd be great!

File: lib/builder/builder.ex [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    "<" <> url <> ">; rel=\"" <> Atom.to_string(relation) <> "\"" <> attrs
+  end
+
+  defp dobuild(_any, _whatever) do
+  end
+
+  defp concatenate(%{} = map , sep, quoted) when is_binary(sep) do
+    Enum.map_join(Map.keys(map), sep, fn(key) ->
+      # TODO: sanitize me (urlencode stuff ?)
```
Comment (simonrand) [inline lib/builder/builder.ex:65]: @xadhoom what's the intention in this TODO?

Comment (xadhoom) [inline lib/builder/builder.ex:65]: well, not really a TODO, I was just wondering if can happen that some k/v pair may contain elements that must be escaped, as the RFC states. Or the escaping is handled by the underlying http library (like put_resp_header pf Plug.Conn) ? 
Right now, if the underlying lib does not escape the header, the user is responsible to pre-escape them.

But didn't had time to investigate it yet.

**Comment by xadhoom** [comment]
Thanks for the merge.
I did not knew about Credo, I'm fairly new to Elixir world. I'll use it for sure, thanks!

**Pull request merged**